src/Microsoft.FeatureManagement.AppInsightsTelemetryPublisher/TelemetryPublisherAppInsights.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.ApplicationInsights;
+using Microsoft.FeatureManagement.Telemetry;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement.AppInsightsTelemetryPublisher
+{
+    /// <summary>
+    /// Used to publish data from evaluation events to Application Insights
+    /// </summary>
+    public class TelemetryPublisherAppInsights : ITelemetryPublisher
+    {
+        private readonly string _eventName = "FeatureEvaluation";
+        private readonly TelemetryClient _telemetryClient;
+
+        public TelemetryPublisherAppInsights(TelemetryClient telemetryClient)
+        {
+            _telemetryClient = telemetryClient;
+        }
+
+        /// <summary>
+        /// Publishes a custom event to Application Insights using data from the given evaluation event.
+        /// </summary>
+        /// <param name="evaluationEvent"> The event to publish.</param>
+        /// <param name="cancellationToken"> A cancellation token.</param>
+        /// <returns>Returns a ValueTask that represents the asynchronous operation</returns>
+        public ValueTask PublishEvent(EvaluationEvent evaluationEvent, CancellationToken cancellationToken)
+        {
+            FeatureDefinition featureDefinition = evaluationEvent.FeatureDefinition;
+
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                { "FeatureName", featureDefinition.Name },
+                { "IsEnabled", evaluationEvent.IsEnabled.ToString() },
+                { "Label", featureDefinition.Label },
+                { "ETag", featureDefinition.ETag },
+            };
+
+            foreach (KeyValuePair<string, string> kvp in featureDefinition.Tags)
+            {
```

Comment:
Seems like app insights has a lesser key length allowance for custom properties than our max allowed tag length. Something to be aware of

https://learn.microsoft.com/en-us/azure/azure-monitor/app/api-custom-events-metrics#limits

Reply:
I tested this, it appears the App Insights sdk automatically substrings(0,150) it when it hits the limit. It will have a key of the first 150 characters. No exceptions are thrown, but we should probably note it in the readme.